dbms/src/Storages/DeltaMerge/LateMaterializationBlockInputStream.cpp

```diff
@@ -112,12 +133,8 @@ Block LateMaterializationBlockInputStream::read()
                 // so only if the number of rows left after filtering out is large enough,
                 // we can skip some packs of the next block, call readWithFilter to get the next block.
                 rest_column_block = rest_column_stream->readWithFilter(*filter);
-                for (auto & col : filter_column_block)
-                {
-                    if (col.name == filter_column_name)
-                        continue;
-                    col.column = col.column->filter(*filter, passed_count);
-                }
+                ColumnPtr filter_column;
```

Comment:
I think we done need to create this `ColumnPtr filter_column` instance before calling `filterFilterColumnBlock`
```suggestion
```